src/main/java/com/elastic/support/diagnostics/commands/RunKibanaQueries.java

```diff
@@ -0,0 +1,309 @@
+            // Get the first actions page
+            queries.add(getNewEntryPage(perPage, 1, action));
+            // If there is more pages add the new queries
+            if (perPage < total) {
+                double numberPages = Math.ceil(total/perPage);
+                for (int i = 2; i <= numberPages; i++) {
+                    queries.add(getNewEntryPage(perPage, i, action));
+                }
+            }
+        }
+    }
+
+    /**
+    * create a new RestEntry page for the original RestEntry action 
+    *
+    * @param  int perPage
+    * @param  int page
+    * @param  RestEntry action
+    * @return RestEntry
+    */
+    private RestEntry getNewEntryPage(double perPage, int page, RestEntry action) {
+        return new RestEntry(String.format("%s_%s", action.getName(), page), "", ".json", false, String.format("%s?per_page=%s&page=%s", action.getUrl(), perPage, page), false);
+    }
+
+
+    /**
+    * This function is executed **after** runBasicQueries
+    * Extract the information on the kibana_node_stats.json with getNodeProfile function.
+    * Within the function getRemoteSystem or getLocalSystem we set ResourceCache.addSystemCommand
+    * Return will be used as workaround to Unit test. Will be replaced in v9
+    *
+    * @param  DiagnosticContext context
+    * @return SystemCommand
+    */
+    public SystemCommand execSystemCommands(DiagnosticContext context) {
+
+        ProcessProfile nodeProfile = getNodeProfile(context.tempDir, "kibana_node_stats.json", context);
+
+        if (StringUtils.isEmpty(nodeProfile.pid) || nodeProfile.pid.equals("1")) {
+            context.dockerPresent = true;
+            context.runSystemCalls = false;
+        }
+        // Create and cache the system command type we need, local or remote...
+        SystemCommand syscmd = null;
+        switch (context.diagnosticInputs.diagType) {
+            case Constants.kibanaRemote:
+                String targetOS;
+                if(context.dockerPresent){
+                    targetOS = Constants.linuxPlatform;
+                }
+                else{
+                    targetOS = nodeProfile.os;
+                }
+                
+                syscmd = getRemoteSystem(targetOS, context);
+                break;
+
+            case Constants.kibanaLocal:
+                if (context.dockerPresent) {
+                    syscmd = getLocalSystem("docker", true);
+                } else {
+                    syscmd = getLocalSystem(nodeProfile.os, false);
+                }
+                break;
+        }
+        return syscmd;
+    }
+
+
+    /**
+    * We will not collect all the headerst hat are returned by the actions API (kibana_actions.json file).
+    * for troubleshooting support engineers will only need "kbn-xsrf" or "Content-Type", all the others are removed.
+    *
+    * @param  DiagnosticContext context
+    * @return void
+    */
+    public void allowedHeadersFilter(DiagnosticContext context) {
```

Comment:
The name of this function is confusing. It sounds generic, but it's not generic at all.

Should this header removal be applied to all Kibana responses? Is headers a special part of the `kibana_actions` API response (and not the literal HTTP headers since the referenced headers _are_ HTTP headers that normally come back from every Kibana API)?

The function should be renamed (probably `filterActionsHeaders` based on reading it) and the answer to those questions should be at least inferable from function comment.